app/lib/text/renderer.rb

```diff
@@ -17,14 +17,27 @@ def render_string(template)
       else
         rendered = template
         rendered = render_report(rendered)
-        @text_component.sensors.each do |sensor|
-          s = SensorDecorator.new(sensor)
-          rendered.gsub!(/({\s*#{ Regexp.quote("value(#{s.id})") }\s*})/, s.last_value(@opts))
+        
+        rendered.scan(/{\s*value\(\s*(\d+)\s*\)\s*}/).each do |sensor_id|
```

Comment:
Maybe you can remove the Regex altogether and replace it with Ruby's format strings: https://ruby-doc.org/core-2.2.0/String.html#method-i-25